src/Dialogue.js

```diff
@@ -26,42 +35,77 @@ export default class Dialogue {
     if (this.isComplete()) {
       return this.completeText;
     } else if (this.isBattle()) {
-      return `${this.conversation[this.progress].description}\n${this.startBattle()}`;
+      return `${this.activeText.description}\n${this.startBattle()}`;
     }
-    return this.conversation[this.progress].description;
+    return this.activeText.description;
   }
 
   /**
    * Restart the dialogue!
    */
   restart() {
-    this.progress = 0;
+    for (var text in this.conversation) {
+      this.activeText = text;
+      break;
+    }
   }
 
   /**
    * For help text, either return its current helpText or its completed text
    * (if complete)
    */
   help() {
-    return this.isComplete() ? this.completeHelp : this.conversation[this.progress].helpText;
+    return this.isComplete() ? this.completeHelp : this.activeText.helpText;
   }
 
   /**
-   * When executing the user's input, check against all possible choices and
-   * return true or false depending on if we have received a match.
+   * Progress the current text state and any associated battle.
+   *
+   * @return - true if there's nothing left to execute, false otherwise.
    */
   execute(input, respond, player) {
     if (this.isBattle()) {
       return this._handleBattle(input, respond, player);
     }
-    return createRegex(this.conversation[this.progress].aliases, false).test(input);
+    return true;//createRegex(this.activeText.aliases, false).test(input);
```

Comment:
This might be what is what's causing items to drop early (if they even are, I don't remember when they were supposed to.
I changed things so the execute function just returns true if execution of the given text block is finished (which at this point pretty much just means any battle is over). I moved the regex creation/testing to a new function.